### ISSUE TITLE
Deleting feeds with all their entries

### ISSUE THREAD

**Issue by andinus** [skeeto/elfeed#392]
Can we delete whole feeds with all their entries? I know I can just tag them`junk' or something but I would like them gone forever. I've read about deleting entries here: https://nullprogram.com/blog/2013/11/26/.

**Comment by shitikanth** [comment]
I disagree with Chris's take here. There is at least one good use case for removing entries (removing the feed entirely). 

Fortunately elfeed's "database" is simple enough to be able to write a bit of elisp to enable removing entries manually. 

```
(defun sk/elfeed-db-remove-entry (id)
  "Removes the entry for ID"
  (avl-tree-delete elfeed-db-index id)
  (remhash id elfeed-db-entries))

(defun sk/elfeed-search-remove-selected ()
  "Remove selected entries from database"
  (interactive)
  (let* ((entries (elfeed-search-selected))
	 (count (length entries)))
    (when (y-or-n-p (format "Delete %d entires?" count))      
      (cl-loop for entry in entries
	       do (sk/elfeed-db-remove-entry (elfeed-entry-id entry)))))
  (elfeed-search-update--force))
```

Just select the entries in the `elfeed-search` buffer that you want to delete and  do `M-x sk/elfeed-search-remove-selected`.

**Comment by slackline** [comment]
Thanks @shitikanth that is really useful.

**Comment by cstanhope** [comment]
Thanks @shitikanth. I realized today a blog I followed changed their feed URL. When I updated the URL, I suddenly had two copies of everything. Not the end of the world, but aesthetically displeasing. Your functions helped me clean up the mess.

**Comment by danielkrajnik** [comment]
"Academics are paid to be smart, practicioners are paid to be right."

`How on earth can you think that not being able to remove entry makes sense in any shape or form` aka I also disagree with Chris's take

**Comment by danielkrajnik** [comment]
To other unfortunate souls that crawled here unsuccessfully trying to remove entries in elfeed-feeds after adding them with `elfeed-add-feed` - it's a "customized variable". You need to remove it via the "customize" dialog.
